pkg/sql/sqlbase/structured.go

```diff
@@ -1198,6 +1201,7 @@ func (desc *MutableTableDescriptor) MaybeFillColumnID(
 	}
 	columnNames[c.Name] = columnID
 	c.ID = columnID
+	c.LogicalColumnID = columnID
```

Comment:
Should this still be done?

Reply:
I think this should still be done, this assigns the column a LogicalColumnID if it's created in an updated version. It makes sense to set it on creation for updated TableDescriptors have this and lazy load/set the LogicalColumnID from previous versions imo.

Reply:
Doing this kind of feels opposite from "only set logical column id if you are swapping columns" or something

Reply:
I don't think "only set logical column id if you are swapping columns" is the right mantra to follow necessarily. That's the only use case right now but not necessarily in the future.

But I don't think LogicalColumnID is specific to swapping columns.

Reply:
sounds good